Added functionality to improve failure triaging

adityapatwardhan opened the pull request:
- Nunit logs will be created to tests.
- Logs are copied to the Azure share.
- Pester is run with -Quiet
- Elevated and unelevated runs for tests with appropriate tags

adityapatwardhan committed:
Added functionality to improve failure triaging

- Nunit logs will be created to tests.
- Logs are copied to the Azure share.
- Pester is run with -Quiet
- Elevated and unelevated runs for tests with appropriate tags

adityapatwardhan committed:
Copy logs to Azure share and upload to CodeCov

- Zip and copy to Azure log share
- Upload to codecov using cygwin

JamesWTruher commented on test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1:
```diff
@@ -99,19 +136,48 @@ try
 
+    $bashScriptInvoker = "$PSScriptRoot\CodecovUploader.cmd"
+    $bashScript = "$PSScriptRoot\CodecovUploader.sh"
+    $cygwinLocation = "$env:SystemDrive\cygwin*"
```
>cygwinLocation [](start = 5, length = 14)

not used?

adityapatwardhan replied:
Fixed. Should have been used in the if condition.

adityapatwardhan committed:
Fixed varaible name in if condition

JamesWTruher commented on test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1:
```diff
@@ -99,19 +136,48 @@ try
+    Invoke-RestMethod 'https://codecov.io/bash' -OutFile $bashScript
+    Write-BashInvokerScript -path $bashScriptInvoker
+
+    if(Test-Path $bashScriptInvoker -and $bashScript -and $cygwinLocation)
```
>t-Path $bashScriptInvoker -and $bashScript -and $cygwinLocation [](start = 10, length = 63)

I think this is:
`if ( test-path $bashScriptInvoker, $bashScript, $cygwinLocation | ?{ ! $_ } ) `

adityapatwardhan replied:
The suggested script works for finding if one of them is false, if all is OK it returns null. Instead fixed by using the following:

```
if((Test-Path $bashScriptInvoker) -and
    (Test-Path $bashScript) -and
    (Test-Path $cygwinLocation)
```

JamesWTruher commented on test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1:
```diff
@@ -14,6 +16,30 @@ function Write-LogPassThru
+    endlocal & %_CYGBIN%\bash --login "%_CYGSCRIPT%" %*
+'@
+
+    $scriptContent | Out-File $path -Force
```
>t-File $path -Force [](start = 23, length = 19)

does this need an explicit encoding?

adityapatwardhan replied:
Does not harm.. i will add encoding ascii

JamesWTruher replied:
if it's not needed, it's fine

---
In reply to: [97681882](https://github.com/PowerShell/PowerShell/pull/3037#discussion_r97681882) [](ancestors = 97681882)

adityapatwardhan committed:
Addressed code review comments

JamesWTruher commented:
:shipit:

adityapatwardhan committed:
Modified the destination folder structure for logs

- Creates a folder structure for yyyy-MM and Windows for storing logs
- Updated zip file name to not have illegal characters

adityapatwardhan committed:
Modifications as suggested by codecov.io

adityapatwardhan commented:
@PowerShell/powershell-maintainers I believe this is ready for merge. Can someone have a look?
